agent/ecscni/plugin_windows.go

```diff
@@ -35,10 +40,32 @@ var (
 		"Amazon", "ECS", "log", "vpc-eni.log")
 )
 
-// setupNS is the called by SetupNS to setup the task namespace by invoking ADD for given CNI configurations
+// setupNS is the called by SetupNS to setup the task namespace by invoking ADD for given CNI configurations.
+// For Windows, we will retry the setup on Windows before conceding error.
 func (client *cniClient) setupNS(ctx context.Context, cfg *Config) (*current.Result, error) {
+	var result *current.Result
+	var err error
+	backoff := retry.NewExponentialBackoff(setupNSBackoffMin, setupNSBackoffMax,
+		setupNSBackoffJitter, setupNSBackoffMultiple)
+
+	for count := 1; count < setupNSMaxRetryCount; count++ {
+		result, err = client.setupNSOnce(ctx, cfg)
+		if err == nil {
+			return result, nil
+		}
+		if count < setupNSMaxRetryCount {
+			time.Sleep(backoff.Duration())
+		}
+		seelog.Debugf("Namespace setup failed. Retry count is %s.", count)
+	}
+	return nil, err
+}
+
+// setupNSOnce invokes the CNI plugins to setup the task network namespace.
+func (client *cniClient) setupNSOnce(ctx context.Context, cfg *Config) (*current.Result, error) {
```

Comment:
nit: I know this naming pattern is present elsewhere, but  I think `doSetupNS` is a more appropriate name for this method.